sample_size/multiple_testing.py

```diff
@@ -0,0 +1,57 @@
+from typing import List
+
+import numpy as np
+from statsmodels.stats.multitest import multipletests
+
+from sample_size.metrics import BaseMetric
+
+
+class MultipleTestingMixin:
+    """
+    This class calculates sample size required under the case of multiple testing
+
+    Attributes:
+    metrics: a list of BaseMetric registered by users
+    variants: number of variants, including control
+    alpha: statistical significance
+    power: statistical power
+
+    """
+
+    metrics: List[BaseMetric]
+    alpha: float
+    power: float
+    variants: int
+
+    def get_multiple_sample_size(self, lower: float, upper: float, depth: int = 0) -> int:
+        MAX_RECURSION_DEPTH = 20
+        EPSILON = 0.025
+
+        if depth > MAX_RECURSION_DEPTH:
+            raise RecursionError
+
+        candidate = int((upper + lower) / 2)
```

Comment:
```suggestion
        candidate = int(upper + (upper-lower) / 2)
```
is upper and lower "size"? maybe try to write it in this way to avoid integer out of bound if it could be a very large number

Reply:
guess we never know if it will be an issue or not? just provide a way so we don't need to worry about that